From f00fbe997a98b9c71d8279bb1753546afd27690c Mon Sep 17 00:00:00 2001
From: auto commit <unknown@unknown>
Date: Fri, 24 Oct 2008 12:23:18 -0700
Subject: [PATCH] cavium: skip the calibrate_delay

Skip the calibrate_delay operation -- all the cores will be
the same by definition.

Signed-off-by: Tomaso Paoletti <tpaoletti@caviumnetworks.com>
Signed-off-by: Paul Gortmaker <Paul.Gortmaker@windriver.com>
---
 arch/mips/kernel/smp.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/arch/mips/kernel/smp.c b/arch/mips/kernel/smp.c
index b214c89..bb1b44b 100644
--- a/arch/mips/kernel/smp.c
+++ b/arch/mips/kernel/smp.c
@@ -116,7 +116,11 @@ asmlinkage __cpuinit void start_secondary(void)
 	 * to an option instead of something based on .cputype
 	 */
 
+#ifndef CONFIG_CPU_CAVIUM_OCTEON
+	/* There is no reason to waste time doing this on Octeon. All the cores
+	    are on the same chip and are the same speed by definition */
 	calibrate_delay();
+#endif
 	preempt_disable();
 	cpu = smp_processor_id();
 	cpu_data[cpu].udelay_val = loops_per_jiffy;
-- 
1.5.5.1

